From: Hannes Reinecke <hare@suse.de>
Date: Wed, 15 Jul 2015 08:16:17 +0200
Subject: scsi_error: add missing case statements in scsi_decide_disposition()
References: bsc#920733
Patch-Mainline: n/a

For virtualized drivers the hostbyte might already be set, but
this doesn't classify as an error. So return SUCCESS here.

Signed-off-by: Hannes Reinecke <hare@suse.de>
---
 drivers/scsi/scsi_error.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/scsi/scsi_error.c b/drivers/scsi/scsi_error.c
index 65f2035..74dce9e 100644
--- a/drivers/scsi/scsi_error.c
+++ b/drivers/scsi/scsi_error.c
@@ -1781,6 +1781,8 @@ int scsi_decide_disposition(struct scsi_cmnd *scmd)
 			return FAILED;
 		}
 	case DID_RESET:
+	case DID_TARGET_FAILURE:
+	case DID_NEXUS_FAILURE:
 		return SUCCESS;
 	default:
 		return FAILED;
-- 
1.8.5.2

